runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6923,6 +6921,11 @@ void TR_J9ByteCodeIlGenerator::genFullFence(TR::Node *node)
 
 void TR_J9ByteCodeIlGenerator::performClassLookahead(TR_PersistentClassInfo *classInfo)
    {
+#if defined(JITSERVER_SUPPORT)
+   // Do not perform class lookahead in server mode
+   if (TR::CompilationInfo::getStream())
```

Comment:
I see that we have the comp object ready so a better test is comp->isOutOfProcessCompilation()